nixos/modules/services/network-filesystems/webdav.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.webdav;
+in
+{
+  options = {
+    services.webdav = {
+      enable = mkEnableOption "WebDAV server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.webdav;
+        description = "The WebDAV package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "User account under which WebDAV runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "Group under which WebDAV runs.";
+      };
+
+      settings = mkOption {
+        type = types.anything;
```

Comment:
```suggestion
        type = format.type;
```

Reply:
In this scenario `format.type` is a nix attribute set and will generate a `yaml` file. See above declaration: `format = pkgs.formats.yaml {};`.